sanic/app.py

```diff
@@ -871,10 +914,18 @@ async def handle_request(self, request: Request):  # no cov
                 if isawaitable(response):
                     response = await response
 
-            if response is not None:
+            if response is not None and not request.responded:
                 response = await request.respond(response)
-            elif not hasattr(handler, "is_websocket"):
-                response = request.stream.response  # type: ignore
+            elif not hasattr(handler, "is_websocket") or request.responded:
```

Comment:
Why in the same conditional?

Reply:
Because I thought connect `request.stream.response` to `response` is also a wanted behavior here when the request was responded (by the response object `request.stream.response`). Would you agree that? Maybe the if else blocks can be better organized?

Reply:
If not connect, test case 4 above may lead to an exception in one of my past experiments. I can reproduce and show that tomorrow.

Reply:
It just seems that you have one conditional and then separate inside. Why not two `elif` one for `is_websocket` check, and the other for the `responded` check?

Reply:
Is this looks better?

```python
            if response is not None:
                if not request.responded:
                    if request.stream is not None:
                        response = request.stream.response
                    error_logger.error(
                        "The response object returned by the route handler "
                        "will not be sent to client. The request has already "
                        "been responded to."
                    )
                else:
                    response = await request.respond(response)
            elif not hasattr(handler, "is_websocket"):
                response = request.stream.response  # type: ignore
```

I wanted to avoid 3 layers if statements so putted the tricky or for set `response` to `request.stream.response`, which should be done in these one of the two conditions.

Reply:
Seems like you are overcomplicating it.

```python
if request.responded:
    error_logger.error(
        "The response object returned by the route handler "
        "will not be sent to client. The request has already "
        "been responded to."
    )
elif response is not None:
    response = await request.respond(response)
elif not hasattr(handler, "is_websocket"):
    response = request.stream.response  # type: ignore
```

Reply:
That's not going to work... For example, error should not be logged if `response is None`, `response = request.stream.response` should also be done (IMO) if `request.responded`. 😂

Reply:
```python
if response is not None and not request.responded:
    response = await request.respond(response)
elif not hasattr(handler, "is_websocket"):
    response = request.stream.response  # type: ignore

if request.responded:
    if request.stream is not None:
        response = request.stream.response if request.stream is not None else None
    error_logger.error(
        "The response object returned by the route handler "
        "will not be sent to client. The request has already "
        "been responded to."
    )
```

This might be better?

Reply:
No, above doesn't work, we have to check `request.responded` before the `respond` call.......

Reply:
Please ignore my messed messages above, sorry about that. The latest push contains modified version of your code. @ahopkins 

Can you do a re-review? Thanks!